src/main.py

```diff
@@ -39,128 +56,170 @@ async def on_close() -> None:
         await bc.finish(say_goodbye=False)
 
 
-@client.event
-async def on_message(message: Message) -> None:
-    global controllers
-    global loaded_image
+# Allow only one async routine to calculate list at a time
+list_task_control_lock = asyncio.Lock()
+
 
-    if message.author == client.user:
+# List command
+@client.slash_command()
+@application_checks.has_role(config.dj_role_name)
+async def list(
+    interaction: Interaction,
+    list_channel: Optional[TextChannel] = SlashOption(required=False),
```

Comment:
I think we can limit `list_channel` to only suggest text channels in Discord's UI by using the `channel_types` argument to `SlashOption`: https://docs.nextcord.dev/en/stable/api.html#nextcord.SlashOption

`list_channel` needs to be of type `GuildChannel` though. So:

```py
list_channel: Optional[GuildChannel] = SlashOption(required=False, channel_types=[ChannelType.text]),
```

I don't know if having `Optional[GuildChannel]` instead of `GuildChannel` for the type will disable the checking code though.

Reply:
Setting the type hint automatically restricts to text channels automatically, [see docs](https://docs.nextcord.dev/en/stable/api.html#slash-options). I personally think it's easier to read as is now than what you suggested, but let me know if you think differently.

I'm also realizing now that the `required=False` argument is also deduced from the `Optional` type. Not sure if we should remove these, lmk what you think.